src/main/java/it/pagopa/pn/user/attributes/middleware/db/v1/ConsentDao.java

```diff
@@ -0,0 +1,107 @@
+package it.pagopa.pn.user.attributes.middleware.db.v1;
+
+import it.pagopa.pn.user.attributes.generated.openapi.server.user.consents.api.v1.dto.ConsentTypeDto;
+import it.pagopa.pn.user.attributes.middleware.db.v1.entities.ConsentEntity;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbAsyncTable;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedAsyncClient;
+import software.amazon.awssdk.enhanced.dynamodb.Key;
+import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
+import software.amazon.awssdk.enhanced.dynamodb.model.GetItemEnhancedRequest;
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional;
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryEnhancedRequest;
+import software.amazon.awssdk.enhanced.dynamodb.model.UpdateItemEnhancedRequest;
+
+@Repository
+@Slf4j
+public class ConsentDao extends BaseDao {
+    DynamoDbAsyncTable<ConsentEntity> userAttributesTable;
+
+    public ConsentDao(DynamoDbEnhancedAsyncClient dynamoDbAsyncClient,
+                      @Value("${pn.user-attributes.dynamodb.table-name}") String table) {
+        this.userAttributesTable = dynamoDbAsyncClient.table(table, TableSchema.fromBean(ConsentEntity.class));
+    }
+
+    // Crea o modifica l'entity ConsentEntity
+
+    /**
+     * Inserice o aggiorna un item di tipo ConsentEntity
+     * setta i campi accepted, created, lastModified
+     * @param userAttributes
+     * @return none
+     */
+    public Mono<Object> consentAction(ConsentEntity userAttributes){
+        GetItemEnhancedRequest getReq = GetItemEnhancedRequest.builder()
+                .key(getKeyBuild(userAttributes.getRecipientId(), userAttributes.getConsentType()))
+                .build();
+
+        return  Mono.fromFuture(userAttributesTable.getItem(getReq).thenApply(r -> {
+                    if (r != null) {
```

Comment:
Non siamo in transazione, tra la get precedente e la update successiva possono avvenire altre modifiche sul DB.

se è stato fatto il ragionamento "L'utente dovrebbe darmi un consenso alla volta e non lavorare con due tab del browser" va segnalato con un commento.